hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -25,30 +296,418 @@ The following table lists the main parameters used throughout the OTP controller
+It is recommended to leave this configuration at the default setting, but SW must under all circumstances write to the {{< regref CHECK_PERIOD_REGEN >}} register in order to lock write access to these CSRs.
+
+Later on during the boot process, SW may also choose to block read access to the {{< regref CREATOR_SW_CFG >}} or {{< regref OWNER_SW_CFG >}} partitions at runtime via {{< regref CREATOR_SW_CFG_READ_LOCK >}} and {{< regref OWNER_SW_CFG_READ_LOCK >}}.
+
+### Reset Considerations
+
+It is important to note that values in OTP **can be corrupted** if a reset occurs during a programming operation.
+This should be of minor concern for SW, however, since all partitions except for the LIFE_CYCLE partition are being provisioned in secure and controlled environments, and not in the field.
+The LIFE_CYCLE partition is the only partition that is modified in the field - but that partition is entirely owned by the life cycle controller and not by SW.
+
+### Programming Already Programmed Regions
+
+OTP words cannot be programmed twice, and doing so may damage the memory array.
+Hence the OTP controller performs a blank check and returns an error if a write operation is issued to an already programmed location.
+
+## Direct Access Interface
+
+OTP has to be programmed via the Direct Access Interface, which is comprised of the following CSRs:
+
+CSR Name                             | Description
+-------------------------------------|------------------------------------
+{{< regref DIRECT_ACCESS_WDATA_0 >}} | Low 32bit word to be written.
+{{< regref DIRECT_ACCESS_WDATA_1 >}} | High 32bit word to be written.
+{{< regref DIRECT_ACCESS_RDATA_0 >}} | Low 32bit word that has been read.
+{{< regref DIRECT_ACCESS_RDATA_1 >}} | High 32bit word that has been read.
+{{< regref DIRECT_ACCESS_ADDRESS >}} | Byte address for the access.
+{{< regref DIRECT_ACCESS_CMD >}}     | Command register to trigger a read or a write access.
+
+### Programming Sequence
+
+A typical programming sequence looks as follows:
+
+1. If the region to be accessed has a 32bit access granule, place a 32bit chunk of data into {{< regref DIRECT_ACCESS_WDATA_0 >}}.
+If the region to be accessed has a 64bit access granule, both the {{< regref DIRECT_ACCESS_WDATA_0 >}} and {{< regref DIRECT_ACCESS_WDATA_1 >}} registers have to be used.
+2. Write the Byte address for the access to {{< regref DIRECT_ACCESS_ADDRESS >}}.
+Note that the address is aligned with the granule, meaning that either 2 or 3 LSBs of the address are ignored, depending on whether the access granule is 32 or 64bit.
+3. Trigger a write command by writing 0x2 {{< regref DIRECT_ACCESS_CMD >}}.
+4. Poll the {{< regref STATUS >}} until the state goes back to IDLE or ERROR.
+Alternatively, the `otp_operation_done` interrupt can be enabled up to notify the processor once an access has completed.
+5. If the operation has finished with ERROR status, additional handling is required (see Section on Error handling).
```

Comment:
maybe add the soft link ofr Error handling? (also applies to line 562)

Reply:
jup good idea - added!